lib/rules/no-common-typos.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @fileoverview Warn for common attribute typos.
+ * @author Koen Punt
+ */
+'use strict';
+
+var Components = require('../util/Components');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+var PATTERN_REGEXP = new RegExp('^/(.+)/(.+)?$');
+
+/**
+ * Convert pattern string to a regular expression.
+ * @param {String|RegExp} string regex
+ * @return {RegExp}
+ */
+function parsePattern(pattern) {
+  if (pattern instanceof RegExp) {
+    return pattern;
+  }
+  if (PATTERN_REGEXP.exec(pattern)) {
+    return new RegExp(RegExp.$1, RegExp.$2);
+  }
+  return new RegExp('^' + pattern + '$');
+}
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Warn for common attribute typos.',
+      category: 'Possible Errors',
+      recommended: false
+    },
+    schema: [{
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          pattern: {
+            type: 'string'
+          },
+          correct: {
+            type: 'string'
+          }
+        },
+        additionalProperties: false
+      }
+    }],
+    fixable: 'code'
+  },
+
+  create: Components.detect(function(context) {
+
+    var rules = context.options || [];
+
+    /**
+     * Checks if we are using an an incorrect attribute name.
+     * @param {ASTNode} node The AST node being checked.
+     * @returns {Boolean} True if an incorrect attribute name is used, false if not.
+     */
+    function isIncorrectAttributeName(regex, correct, node) {
+      return Boolean(
+        node.type === 'JSXAttribute' &&
+        node.name &&
+        node.name.name !== correct &&
+        regex.test(node.name.name)
+      );
+    }
+
+    return {
+      JSXAttribute: function(node) {
+        rules.forEach(function(rule) {
+          var regex = parsePattern(rule.pattern);
```

Comment:
I changed it already so that regular expressions are always treated as string. Updated my previous comment to reflect that.